Sources/NIO/BaseSocketChannel.swift

```diff
@@ -755,9 +751,36 @@ class BaseSocketChannel<T: BaseSocket>: SelectableChannel, ChannelCore {
         }
     }
 
+    final func readEOF() {
+        if self.lifecycleManager.isRegistered {
+            self.safeReregister(interested: self.interestedEvent.subtracting(.readEOF))
```

Comment:
Can we add a comment to say why we subtract `readEOF` (namely, that we want this to be one-shot).